demo_spaceros/demo_spaceros.repos

```diff
@@ -0,0 +1,274 @@
+repositories:
```

Comment:
This repos file looks overprovisioned as it is including things which are explicitly ignored in the upstream spaceros dependencies.

Now that the primary space-ros repositories file has switched to a generated approach (https://github.com/space-ros/space-ros/pull/43) it may make sense to try and use a similar approach here and use rosinstall_generator during the docker build to "fill in" dependencies between the Space ROS core and the demos packages. We would have to prove that rosinstall_generator's `--exclude-path` argument can work with install spaces in addition to source spaces but _I think_ it should as long as the installed packages correctly install their package.xml files. Since `--exclude-path` works recursively it also means we ought not need to propagate forward the `excluded-pkgs.txt` content unless those packages are also direct dependencies of packages that aren't already installed.

Reply:
There are still a significant number of extras which ought not be part of the build as far as I can tell. How did you generate this repos file?